runtime/oti/j9consts.h

```diff
@@ -732,15 +732,16 @@ extern "C" {
 #define J9_JNI_OFFLOAD_SWITCH_JVMTI_GC_FINISH							0x36
 #define J9_JNI_OFFLOAD_SWITCH_JVMTI_OBJECT_FREE							0x37
 #define J9_JNI_OFFLOAD_SWITCH_JVMTI_VM_OBJECT_ALLOC						0x38
-#define J9_JNI_OFFLOAD_SWITCH_J9JVMTI_COMPILING_START					0x39
-#define J9_JNI_OFFLOAD_SWITCH_J9JVMTI_COMPILING_END						0x3A
-#define J9_JNI_OFFLOAD_SWITCH_J9JVMTI_INSTRUMENTABLE_OBJECT_ALLOC		0x3B
-#define J9_JNI_OFFLOAD_SWITCH_J9JVMTI_VM_DUMP_START						0x3C
-#define J9_JNI_OFFLOAD_SWITCH_J9JVMTI_VM_DUMP_END						0x3D
-#define J9_JNI_OFFLOAD_SWITCH_J9JVMTI_GC_CYCLE_START					0x3E
-#define J9_JNI_OFFLOAD_SWITCH_J9JVMTI_GC_CYCLE_FINISH					0x3F
-#define J9_JNI_OFFLOAD_SWITCH_TRACE_SUBSCRIBER_THREAD					0x40
-#define J9_JNI_OFFLOAD_SWITCH_JVMTI_SAMPLED_OBJECT_ALLOC				0x41
+/* 0x39 Reserved for JVMTI event */
```

Comment:
why the renumbering and why is 0x39 reserved?

Reply:
> why the renumbering and why is 0x39 reserved?

There are two kind of JVMTI events, one is from the JVMTI spec and defined at https://github.com/eclipse/openj9/blob/00bc8b44faf3cd10181871f3d43f98f94b41c7e9/runtime/include/jvmti.h.m4#L738-L741
the other is OpenJ9 specific JVMTI events and defined at https://github.com/eclipse/openj9/blob/00bc8b44faf3cd10181871f3d43f98f94b41c7e9/runtime/oti/jvmtiInternal.h#L36-L48
The OpenJ9 internal ones always starts from `J9JVMTI_HIGHEST_EVENT` and are larger than all JVMTI spec event defines.

`javaOffloadSwitchOn()` & `javaOffloadSwitchOff()` use `eventNumber - J9JVMTI_LOWEST_EVENT` to determine the reason code.
https://github.com/eclipse/openj9/blob/00bc8b44faf3cd10181871f3d43f98f94b41c7e9/runtime/jvmti/jvmtiHelpers.c#L631-L632

Hence this new `JDK11` `J9_JNI_OFFLOAD_SWITCH_JVMTI_SAMPLED_OBJECT_ALLOC` has to match its event define within `jvmti.h`, i.e., it need to be inserted between previous `JDK 8` and `OpenJ9` specific defines.